wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -959,9 +995,45 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
             preDownloadBlocks.set(true)
             preDownloadBlocksFuture = future
             log.info("PreDownloadBlocks: starting")
+
+            //first check to see if there is a blockchain identity
+            if (blockchainIdentityDataDao.load() == null) {
+                log.info("PreDownloadBlocks: checking for existing associated identity")
+
+                val identity = getIdentityFromPublicKeyId()
+                if (identity != null) {
+                    log.info("PreDownloadBlocks: initiate recovery of existing identity ${identity.id}")
+                    ContextCompat.startForegroundService(walletApplication, createIntentForRestore(walletApplication, identity.id))
+                    return@launch
+                } else {
+                    log.info("PreDownloadBlocks: no existing identity found")
+                }
+            }
+
+            // update contacts, profiles and other platform data
             if (!updatingContacts.get()) {
                 updateContactRequests()
             }
         }
     }
+
+    /**
+    This is used by java code, outside of coroutines
+
+    This should not be a suspended method.
+     */
+    fun clearDatabases() {
```

Comment:
Shouldn't this be called `clearDatabase` (in singular) or `clearTables`?

Reply:
yes, I will change this to `clearDatabase()`